docs/development/standards.rst

```diff
@@ -1,159 +1,75 @@
-        installed via `Bower`_ and included in the output library file. Because
-        we aren't reusing them commonly, they don't require a separate bundle or
-        separate include. Examples here would include jQuery plugins used on one
-        off forms, such as jQuery Payments.
-
-    Our libraries
-        These libraries are bundled up excluding vendor libraries ignored by
-        rules in our ``gulpfile.js``. These files should be organized by
-        function and can be split up into multiple files per application.
-
-        Entry points to libraries must be defined in ``gulpfile.js`` for now. We
-        don't have a defined directory structure that would make it easy to
-        imply the entry point to an application library.
-
-.. _`Bower`: http://bower.io
-.. _`Gulp`: http://gulpjs.com
-.. _`Browserify`: http://browserify.org
-.. _`Debowerify`: https://github.com/eugeneware/debowerify
-.. _`LESS`: http://lesscss.org
 
-.. _`jQuery`: http://jquery.com
-.. _`Knockout`: http://knockoutjs.com
+These are build standards that are adhered to by the core development team while
+developing Read the Docs and related services. If you are a new contributor to
+Read the Docs, it might a be a good idea to follow these guidelines as well. The
+:doc:`standard installation instructions <../install>` do cover what it takes to
+get started with a local installation of Read the Docs and can be used for local
+development by non-core team.
+
+Core team standards
+-------------------
+
+Core team members expect to have a development environment that closely
+approximates our production environment, in order to spot bugs and logical
+inconsistencies before they make their way to production.
+
+Celery runs as a separate process
+    Celery needs to be run as a separate process, and core team will not use
+    in-process task execution via Celery's ``ALWAYS_EAGER`` setting. The
+    ``ALWAYS_EAGER`` setting doesn't work correctly, and masks bugs that
+    introduce task execution race conditions.
+
+Celery runs multiple processes
+    We run celery with multiple worker processes to discover race conditions
+    between tasks.
+
+Docker for builds
+    Docker is used for a build backend instead of the local host build backend.
+    There are a number of differences between the two execution methods in how
+    processes are executed, what is installed, and what can potentially leak
+    through and mask bugs -- for example, local SSH agent allowing code check
+    not normally possible.
+
+Setting ``USE_SUBDOMAIN = True``
+    There are a number of resolution bugs and cross-domain behavior that can
+    only be caught by using this setting. It requires use of a catch all public
+    domain, so we can resolve all subdomains to ``localhost``. For example, here
+    are some related settings that affect project URLs::
+
+        PRODUCTION_DOMAIN = 'localhost:8000'
+        PUBLIC_DOMAIN = 'dev.readthedocs.io:8001'
+        USE_SUBDOMAIN = True
```

Comment:
If we are going to make recommendations around this, we should consider making the preferred defaults live in `settings/base.py` or having a `settings/local_settings_example.py`

Reply:
We can address this separately. I think a core settings file for web/build makes sense.